agents/labjack/labjack_agent.py

```diff
@@ -342,6 +349,104 @@ def stop_acq(self, session, params=None):
+        scan_rate_input = params.get('sampling_frequency',
+                                     self.sampling_frequency)
+        #Warn user that they input too fast of a sample rate and set
+        #to maximum allowable.
+        if scan_rate_input > max_fsamp:
+            self.log.warn(f'Sampling rate {scan_rate_input} exceeds'
+                          'allowable range for register read mode'
+                          'if you want to sample faster please use'
+                          'standard acquire mode and do not read out '
+                          'any non-standard channels. Setting sampling'
+                          f'rate to maximum = {max_fsamp} Hz')
+            scan_rate_input = max_fsamp
+            scan_rate_dt = 0
+        else:
+            scan_rate_dt = (1/scan_rate_input) - read_dt
+            
+        scans_per_read = max(1, int(scan_rate_input))
+        with self.lock.acquire_timeout(0, job='acq') as acquired:
+            if not acquired:
+                self.log.warn("Could not start acq because "
+                              "{} is already running".format(self.lock.job))
+                return False, "Could not acquire lock."
+
+            session.set_status('running')
+            self.take_data = True
+
+            while self.take_data:
+                data = {
+                    'block_name': 'reg',
+                    'data': {}
+                }
+
+                # Get a timestamp
+                timestamp = time.time()
+                # Query the labjack
+                output = ljm.eReadNames(self.handle,num_chs,self.chs)
+
+                # Data comes in form ['reg1', 'reg2', 'reg3', ...]
+                # where regn are the registers in your active_channels
+                # list in your site config file.
+                for i, ch in enumerate(self.chs):
+                    ch_output = output[i]
+                    data['data'][ch] = ch_output
+
+                    # Apply unit conversion function for this channel
+                    if ch in self.functions.keys():
+                        new_ch_output, units = \
+                            self.ljf.unit_conversion(ch_output,
+                                                     self.functions[ch])
+                        data['data'][ch + units] = new_ch_output
+
+                data['timestamp'] = timestamp
+                self.agent.publish_to_feed('registers', data)
+                time.sleep(scan_rate_dt)
+
+            # Flush buffer and stop the data stream
+            self.agent.feeds['registers'].flush_buffer()
+            ljm.close(self.handle)
+            self.log.info("Data stream stopped")
+        return True, 'Acquisition exited cleanly.'
+
+    def stop_acq_reg(self, session, params=None):
+        if self.take_data:
+            self.take_data = False
+            return True, 'requested to stop taking data.'
+        else:
+            return False, 'acq is not currently running'
```

Comment:
This is exactly duplicating `stop_acq`, since `self.take_data` is used to track when either is running. I wonder if we can just use `stop_acq` in both cases. That would involve removing this, and updating the `register_process` lines on 506 and 507 to use `sensors.stop_acq` instead.

Reply:
Switched to just stop_acq